MySQL init process hangs when using image mariadb 10.1.42, 10.2.27, 10.3.18, 10.4.8

I am using `mariadb:10.3`, and there seems to be an issue with the current version (10.3 == 10.3.18 == sha256:17a2a44fb692a7a4b244b104de0f286b51b29083b794ae964e3ed5aab8259358), where the MySQL init process is stuck when running a "fresh" docker (I mean, with no existing database in a volume).

For example, with 10.3.18 the output looks like this:

```
docker run -e MYSQL_ROOT_PASSWORD=wut mariadb:10.3.18
[...]
MySQL init process in progress...
[...]
Version: '10.3.18-MariaDB-1:10.3.18+maria~bionic'  socket: '/var/run/mysqld/mysqld.sock'  port: 0  mariadb.org binary distribution
# stuck here
```

Whereas, with 10.3.17, the output looks like this:

```
docker run -e MYSQL_ROOT_PASSWORD=wut mariadb:10.3.17
[...]
MySQL init process in progress...
[...]
MySQL init process done. Ready for start up.
[...]
# life goes on
```

Here is the full output when using 10.3.18:
```
$ docker run -e MYSQL_ROOT_PASSWORD=wut mariadb:10.3.18
Initializing database


PLEASE REMEMBER TO SET A PASSWORD FOR THE MariaDB root USER !
To do so, start the server, then issue the following commands:

'/usr/bin/mysqladmin' -u root password 'new-password'
'/usr/bin/mysqladmin' -u root -h  password 'new-password'

Alternatively you can run:
'/usr/bin/mysql_secure_installation'

which will also give you the option of removing the test
databases and anonymous user created by default.  This is
strongly recommended for production servers.

See the MariaDB Knowledgebase at http://mariadb.com/kb or the
MySQL manual for more instructions.

Please report any problems at http://mariadb.org/jira

The latest information about MariaDB is available at http://mariadb.org/.
You can find additional information about the MySQL part at:
http://dev.mysql.com
Consider joining MariaDB's strong and vibrant community:
https://mariadb.org/get-involved/

Database initialized
MySQL init process in progress...
2019-09-20 14:31:51 0 [Note] mysqld (mysqld 10.3.18-MariaDB-1:10.3.18+maria~bionic) starting as process 109 ...
2019-09-20 14:31:51 0 [Note] InnoDB: Using Linux native AIO
2019-09-20 14:31:51 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2019-09-20 14:31:51 0 [Note] InnoDB: Uses event mutexes
2019-09-20 14:31:51 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2019-09-20 14:31:51 0 [Note] InnoDB: Number of pools: 1
2019-09-20 14:31:51 0 [Note] InnoDB: Using SSE2 crc32 instructions
2019-09-20 14:31:51 0 [Note] InnoDB: Initializing buffer pool, total size = 256M, instances = 1, chunk size = 128M
2019-09-20 14:31:51 0 [Note] InnoDB: Completed initialization of buffer pool
2019-09-20 14:31:51 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
2019-09-20 14:31:51 0 [Note] InnoDB: 128 out of 128 rollback segments are active.
2019-09-20 14:31:51 0 [Note] InnoDB: Creating shared tablespace for temporary tables
2019-09-20 14:31:51 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2019-09-20 14:31:51 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
2019-09-20 14:31:51 0 [Note] InnoDB: 10.3.18 started; log sequence number 1630860; transaction id 21
2019-09-20 14:31:51 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
2019-09-20 14:31:51 0 [Note] Plugin 'FEEDBACK' is disabled.
2019-09-20 14:31:51 0 [Note] InnoDB: Buffer pool(s) load completed at 190920 14:31:51
2019-09-20 14:31:51 0 [Warning] 'user' entry 'root@bfd3fc90f6f5' ignored in --skip-name-resolve mode.
2019-09-20 14:31:51 0 [Warning] 'user' entry '@bfd3fc90f6f5' ignored in --skip-name-resolve mode.
2019-09-20 14:31:51 0 [Warning] 'proxies_priv' entry '@% root@bfd3fc90f6f5' ignored in --skip-name-resolve mode.
2019-09-20 14:31:51 0 [Note] Reading of all Master_info entries succeeded
2019-09-20 14:31:51 0 [Note] Added new Master_info '' to hash table
2019-09-20 14:31:51 0 [Note] mysqld: ready for connections.
Version: '10.3.18-MariaDB-1:10.3.18+maria~bionic'  socket: '/var/run/mysqld/mysqld.sock'  port: 0  mariadb.org binary distribution
Warning: Unable to load '/usr/share/zoneinfo/leap-seconds.list' as time zone. Skipping it.
```

@yosifkit

> The multiple workarounds are:
> 
> 1. use a faster disk
> 2. `MYSQL_INITDB_SKIP_TZINFO=1` and add tzdata yourself 
...

What do you mean by 

> add tzdata yourself ?

Running 

`mysql_tzinfo_to_sql /usr/share/zoneinfo | sed 's/Local time zone must be set--see zic manual page/FCTY/' | mysql -uuser -ppass --database=mysql` 

similarly to `docker-entrypoint.sh`?

I ran the above commands against one node after MariaDB cluster of 3 nodes went up successfully.
Execution of commands took ~12 minutes and they finished with warning: 

`Warning: Unable to load '/usr/share/zoneinfo/leap-seconds.list' as time zone. Skipping it.` 

In addition, there were errors in the logs of other 2 nodes in the cluster like:

`[ERROR] Slave SQL: Error 'Column 'Time_zone_id' cannot be null' on query. Default database: 'mysql'. Query: 'INSERT INTO time_zone_name (Name, Time_zone_id) VALUES ('Africa/Abidjan', @time_zone_id)', Internal MariaDB error code: 1048`

--- a/10.1/docker-entrypoint.sh
+++ b/10.1/docker-entrypoint.sh
@@ -205,11 +205,21 @@ docker_process_sql() {
 docker_setup_db() {
 	# Load timezone info into database
 	if [ -z "$MYSQL_INITDB_SKIP_TZINFO" ]; then
+		# Aria in 10.4+ is slow due to "transactional" (crash safety)
+		# https://jira.mariadb.org/browse/MDEV-23326 / https://github.com/docker-library/mariadb/issues/262
+		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
+		do
+			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
+		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
 		# sed is for https://bugs.mysql.com/bug.php?id=20545
 		mysql_tzinfo_to_sql /usr/share/zoneinfo \
 			| sed 's/Local time zone must be set--see zic manual page/FCTY/' \
 			| docker_process_sql --dont-use-mysql-root-password --database=mysql
 			# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
+		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
+		do
+			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
+		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
 	fi
 	# Generate random root password
 	if [ -n "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then

--- a/10.1/docker-entrypoint.sh
+++ b/10.1/docker-entrypoint.sh
@@ -205,21 +205,24 @@ docker_process_sql() {
 docker_setup_db() {
 	# Load timezone info into database
 	if [ -z "$MYSQL_INITDB_SKIP_TZINFO" ]; then
-		# Aria in 10.4+ is slow due to "transactional" (crash safety)
-		# https://jira.mariadb.org/browse/MDEV-23326 / https://github.com/docker-library/mariadb/issues/262
-		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
-		do
-			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
-		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
-		# sed is for https://bugs.mysql.com/bug.php?id=20545
-		mysql_tzinfo_to_sql /usr/share/zoneinfo \
-			| sed 's/Local time zone must be set--see zic manual page/FCTY/' \
-			| docker_process_sql --dont-use-mysql-root-password --database=mysql
-			# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
-		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
-		do
-			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
-		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
+		{
+			# Aria in 10.4+ is slow due to "transactional" (crash safety)
+			# https://jira.mariadb.org/browse/MDEV-23326
+			# https://github.com/docker-library/mariadb/issues/262
+			local tztables=( time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type )
+			for table in "${tztables[@]}"; do
+				echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
+			done
+
+			# sed is for https://bugs.mysql.com/bug.php?id=20545
+			mysql_tzinfo_to_sql /usr/share/zoneinfo \
+				| sed 's/Local time zone must be set--see zic manual page/FCTY/'
+
+			for table in "${tztables[@]}"; do
+				echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
+			done
+		} | docker_process_sql --dont-use-mysql-root-password --database=mysql
+		# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
 	fi
 	# Generate random root password
 	if [ -n "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then

--- a/10.2/docker-entrypoint.sh
+++ b/10.2/docker-entrypoint.sh
@@ -205,11 +205,21 @@ docker_process_sql() {
 docker_setup_db() {
 	# Load timezone info into database
 	if [ -z "$MYSQL_INITDB_SKIP_TZINFO" ]; then
+		# Aria in 10.4+ is slow due to "transactional" (crash safety)
+		# https://jira.mariadb.org/browse/MDEV-23326 / https://github.com/docker-library/mariadb/issues/262
+		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
+		do
+			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
+		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
 		# sed is for https://bugs.mysql.com/bug.php?id=20545
 		mysql_tzinfo_to_sql /usr/share/zoneinfo \
 			| sed 's/Local time zone must be set--see zic manual page/FCTY/' \
 			| docker_process_sql --dont-use-mysql-root-password --database=mysql
 			# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
+		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
+		do
+			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
+		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
 	fi
 	# Generate random root password
 	if [ -n "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then

--- a/10.2/docker-entrypoint.sh
+++ b/10.2/docker-entrypoint.sh
@@ -205,21 +205,24 @@ docker_process_sql() {
 docker_setup_db() {
 	# Load timezone info into database
 	if [ -z "$MYSQL_INITDB_SKIP_TZINFO" ]; then
-		# Aria in 10.4+ is slow due to "transactional" (crash safety)
-		# https://jira.mariadb.org/browse/MDEV-23326 / https://github.com/docker-library/mariadb/issues/262
-		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
-		do
-			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
-		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
-		# sed is for https://bugs.mysql.com/bug.php?id=20545
-		mysql_tzinfo_to_sql /usr/share/zoneinfo \
-			| sed 's/Local time zone must be set--see zic manual page/FCTY/' \
-			| docker_process_sql --dont-use-mysql-root-password --database=mysql
-			# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
-		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
-		do
-			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
-		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
+		{
+			# Aria in 10.4+ is slow due to "transactional" (crash safety)
+			# https://jira.mariadb.org/browse/MDEV-23326
+			# https://github.com/docker-library/mariadb/issues/262
+			local tztables=( time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type )
+			for table in "${tztables[@]}"; do
+				echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
+			done
+
+			# sed is for https://bugs.mysql.com/bug.php?id=20545
+			mysql_tzinfo_to_sql /usr/share/zoneinfo \
+				| sed 's/Local time zone must be set--see zic manual page/FCTY/'
+
+			for table in "${tztables[@]}"; do
+				echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
+			done
+		} | docker_process_sql --dont-use-mysql-root-password --database=mysql
+		# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
 	fi
 	# Generate random root password
 	if [ -n "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then

--- a/10.3/docker-entrypoint.sh
+++ b/10.3/docker-entrypoint.sh
@@ -205,11 +205,21 @@ docker_process_sql() {
 docker_setup_db() {
 	# Load timezone info into database
 	if [ -z "$MYSQL_INITDB_SKIP_TZINFO" ]; then
+		# Aria in 10.4+ is slow due to "transactional" (crash safety)
+		# https://jira.mariadb.org/browse/MDEV-23326 / https://github.com/docker-library/mariadb/issues/262
+		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
+		do
+			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
+		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
 		# sed is for https://bugs.mysql.com/bug.php?id=20545
 		mysql_tzinfo_to_sql /usr/share/zoneinfo \
 			| sed 's/Local time zone must be set--see zic manual page/FCTY/' \
 			| docker_process_sql --dont-use-mysql-root-password --database=mysql
 			# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
+		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
+		do
+			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
+		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
 	fi
 	# Generate random root password
 	if [ -n "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then

--- a/10.3/docker-entrypoint.sh
+++ b/10.3/docker-entrypoint.sh
@@ -205,21 +205,24 @@ docker_process_sql() {
 docker_setup_db() {
 	# Load timezone info into database
 	if [ -z "$MYSQL_INITDB_SKIP_TZINFO" ]; then
-		# Aria in 10.4+ is slow due to "transactional" (crash safety)
-		# https://jira.mariadb.org/browse/MDEV-23326 / https://github.com/docker-library/mariadb/issues/262
-		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
-		do
-			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
-		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
-		# sed is for https://bugs.mysql.com/bug.php?id=20545
-		mysql_tzinfo_to_sql /usr/share/zoneinfo \
-			| sed 's/Local time zone must be set--see zic manual page/FCTY/' \
-			| docker_process_sql --dont-use-mysql-root-password --database=mysql
-			# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
-		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
-		do
-			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
-		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
+		{
+			# Aria in 10.4+ is slow due to "transactional" (crash safety)
+			# https://jira.mariadb.org/browse/MDEV-23326
+			# https://github.com/docker-library/mariadb/issues/262
+			local tztables=( time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type )
+			for table in "${tztables[@]}"; do
+				echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
+			done
+
+			# sed is for https://bugs.mysql.com/bug.php?id=20545
+			mysql_tzinfo_to_sql /usr/share/zoneinfo \
+				| sed 's/Local time zone must be set--see zic manual page/FCTY/'
+
+			for table in "${tztables[@]}"; do
+				echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
+			done
+		} | docker_process_sql --dont-use-mysql-root-password --database=mysql
+		# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
 	fi
 	# Generate random root password
 	if [ -n "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then

--- a/10.4/docker-entrypoint.sh
+++ b/10.4/docker-entrypoint.sh
@@ -205,11 +205,21 @@ docker_process_sql() {
 docker_setup_db() {
 	# Load timezone info into database
 	if [ -z "$MYSQL_INITDB_SKIP_TZINFO" ]; then
+		# Aria in 10.4+ is slow due to "transactional" (crash safety)
+		# https://jira.mariadb.org/browse/MDEV-23326 / https://github.com/docker-library/mariadb/issues/262
+		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
+		do
+			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
+		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
 		# sed is for https://bugs.mysql.com/bug.php?id=20545
 		mysql_tzinfo_to_sql /usr/share/zoneinfo \
 			| sed 's/Local time zone must be set--see zic manual page/FCTY/' \
 			| docker_process_sql --dont-use-mysql-root-password --database=mysql
 			# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
+		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
+		do
+			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
+		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
 	fi
 	# Generate random root password
 	if [ -n "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then

--- a/10.4/docker-entrypoint.sh
+++ b/10.4/docker-entrypoint.sh
@@ -205,21 +205,24 @@ docker_process_sql() {
 docker_setup_db() {
 	# Load timezone info into database
 	if [ -z "$MYSQL_INITDB_SKIP_TZINFO" ]; then
-		# Aria in 10.4+ is slow due to "transactional" (crash safety)
-		# https://jira.mariadb.org/browse/MDEV-23326 / https://github.com/docker-library/mariadb/issues/262
-		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
-		do
-			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
-		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
-		# sed is for https://bugs.mysql.com/bug.php?id=20545
-		mysql_tzinfo_to_sql /usr/share/zoneinfo \
-			| sed 's/Local time zone must be set--see zic manual page/FCTY/' \
-			| docker_process_sql --dont-use-mysql-root-password --database=mysql
-			# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
-		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
-		do
-			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
-		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
+		{
+			# Aria in 10.4+ is slow due to "transactional" (crash safety)
+			# https://jira.mariadb.org/browse/MDEV-23326
+			# https://github.com/docker-library/mariadb/issues/262
+			local tztables=( time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type )
+			for table in "${tztables[@]}"; do
+				echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
+			done
+
+			# sed is for https://bugs.mysql.com/bug.php?id=20545
+			mysql_tzinfo_to_sql /usr/share/zoneinfo \
+				| sed 's/Local time zone must be set--see zic manual page/FCTY/'
+
+			for table in "${tztables[@]}"; do
+				echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
+			done
+		} | docker_process_sql --dont-use-mysql-root-password --database=mysql
+		# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
 	fi
 	# Generate random root password
 	if [ -n "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then

--- a/10.5/docker-entrypoint.sh
+++ b/10.5/docker-entrypoint.sh
@@ -205,11 +205,21 @@ docker_process_sql() {
 docker_setup_db() {
 	# Load timezone info into database
 	if [ -z "$MYSQL_INITDB_SKIP_TZINFO" ]; then
+		# Aria in 10.4+ is slow due to "transactional" (crash safety)
+		# https://jira.mariadb.org/browse/MDEV-23326 / https://github.com/docker-library/mariadb/issues/262
+		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
+		do
+			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
+		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
 		# sed is for https://bugs.mysql.com/bug.php?id=20545
 		mysql_tzinfo_to_sql /usr/share/zoneinfo \
 			| sed 's/Local time zone must be set--see zic manual page/FCTY/' \
 			| docker_process_sql --dont-use-mysql-root-password --database=mysql
 			# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
+		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
+		do
+			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
+		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
 	fi
 	# Generate random root password
 	if [ -n "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then

--- a/10.5/docker-entrypoint.sh
+++ b/10.5/docker-entrypoint.sh
@@ -205,21 +205,24 @@ docker_process_sql() {
 docker_setup_db() {
 	# Load timezone info into database
 	if [ -z "$MYSQL_INITDB_SKIP_TZINFO" ]; then
-		# Aria in 10.4+ is slow due to "transactional" (crash safety)
-		# https://jira.mariadb.org/browse/MDEV-23326 / https://github.com/docker-library/mariadb/issues/262
-		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
-		do
-			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
-		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
-		# sed is for https://bugs.mysql.com/bug.php?id=20545
-		mysql_tzinfo_to_sql /usr/share/zoneinfo \
-			| sed 's/Local time zone must be set--see zic manual page/FCTY/' \
-			| docker_process_sql --dont-use-mysql-root-password --database=mysql
-			# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
-		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
-		do
-			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
-		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
+		{
+			# Aria in 10.4+ is slow due to "transactional" (crash safety)
+			# https://jira.mariadb.org/browse/MDEV-23326
+			# https://github.com/docker-library/mariadb/issues/262
+			local tztables=( time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type )
+			for table in "${tztables[@]}"; do
+				echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
+			done
+
+			# sed is for https://bugs.mysql.com/bug.php?id=20545
+			mysql_tzinfo_to_sql /usr/share/zoneinfo \
+				| sed 's/Local time zone must be set--see zic manual page/FCTY/'
+
+			for table in "${tztables[@]}"; do
+				echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
+			done
+		} | docker_process_sql --dont-use-mysql-root-password --database=mysql
+		# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
 	fi
 	# Generate random root password
 	if [ -n "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then

--- a/docker-entrypoint.sh
+++ b/docker-entrypoint.sh
@@ -205,11 +205,21 @@ docker_process_sql() {
 docker_setup_db() {
 	# Load timezone info into database
 	if [ -z "$MYSQL_INITDB_SKIP_TZINFO" ]; then
+		# Aria in 10.4+ is slow due to "transactional" (crash safety)
+		# https://jira.mariadb.org/browse/MDEV-23326 / https://github.com/docker-library/mariadb/issues/262
+		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
+		do
+			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
+		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
 		# sed is for https://bugs.mysql.com/bug.php?id=20545
 		mysql_tzinfo_to_sql /usr/share/zoneinfo \
 			| sed 's/Local time zone must be set--see zic manual page/FCTY/' \
 			| docker_process_sql --dont-use-mysql-root-password --database=mysql
 			# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
+		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
+		do
+			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
+		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
 	fi
 	# Generate random root password
 	if [ -n "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then

--- a/docker-entrypoint.sh
+++ b/docker-entrypoint.sh
@@ -205,21 +205,24 @@ docker_process_sql() {
 docker_setup_db() {
 	# Load timezone info into database
 	if [ -z "$MYSQL_INITDB_SKIP_TZINFO" ]; then
-		# Aria in 10.4+ is slow due to "transactional" (crash safety)
-		# https://jira.mariadb.org/browse/MDEV-23326 / https://github.com/docker-library/mariadb/issues/262
-		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
-		do
-			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
-		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
-		# sed is for https://bugs.mysql.com/bug.php?id=20545
-		mysql_tzinfo_to_sql /usr/share/zoneinfo \
-			| sed 's/Local time zone must be set--see zic manual page/FCTY/' \
-			| docker_process_sql --dont-use-mysql-root-password --database=mysql
-			# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
-		for table in time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type
-		do
-			echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
-		done| docker_process_sql --dont-use-mysql-root-password --database=mysql
+		{
+			# Aria in 10.4+ is slow due to "transactional" (crash safety)
+			# https://jira.mariadb.org/browse/MDEV-23326
+			# https://github.com/docker-library/mariadb/issues/262
+			local tztables=( time_zone time_zone_leap_second time_zone_name time_zone_transition time_zone_transition_type )
+			for table in "${tztables[@]}"; do
+				echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=0 */;"
+			done
+
+			# sed is for https://bugs.mysql.com/bug.php?id=20545
+			mysql_tzinfo_to_sql /usr/share/zoneinfo \
+				| sed 's/Local time zone must be set--see zic manual page/FCTY/'
+
+			for table in "${tztables[@]}"; do
+				echo "/*!100400 ALTER TABLE $table TRANSACTIONAL=1 */;"
+			done
+		} | docker_process_sql --dont-use-mysql-root-password --database=mysql
+		# tell docker_process_sql to not use MYSQL_ROOT_PASSWORD since it is not set yet
 	fi
 	# Generate random root password
 	if [ -n "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then

